internal/services/containers/kubernetes_cluster_resource.go

```diff
@@ -2281,6 +2305,19 @@ func resourceKubernetesClusterUpdate(d *pluginsdk.ResourceData, meta interface{}
 		}
 	}
 
+	if d.HasChange("upgrade_override") {
+		upgradeOverrideSettingRaw := d.Get("upgrade_override").([]interface{})
+
+		err = validateKubernetesClusterUpgradeOverrideSetting(d.GetChange("upgrade_override"))
```

Comment:
Would it make more sense to put this logic inline rather than in a helper function to be more consistent with the other resources in the provider? I might be wrong but I believe this is the only place this function is called, so to me it would make more sense to make that logic inline, here in the `update` fuction, to be more readable for future maintainers is they need to modify this code.